<a href="https://colab.research.google.com/github/bngrossmann/new-york-crime-analysis/blob/main/New_York_Crime_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2

* Benjamin Grossmann

# Part 1


Your second project is going to have a lot more freedom than your first project. This is because we want you to have a project in your portfolio that interests you or relates to the industry you would like to work in.

Your task for this week is to propose two possible datasets you would like to work with for Project 2.  

You will choose your first choice data set, and a backup data set in case the first proposed data set is not approved.  

This data can be from any source and can be on any topic with these limitations:

the data must be available for use (it is your responsibility to ensure that the license states that you are able to use it.)
the data must be appropriate for a professional environment
the data must NOT contain personal information
the data must NOT be a dataset used for any assignment, lecture, or task from the course
Make sure you select a dataset that will be reasonable to work with in the amount of time we have left. Think about what questions you could reasonably answer with the dataset you select. 

You must propose two datasets that each have a supervised learning component. You may choose a regression or classification problem for each proposed data set.

**For this task:**

Create a Colab notebook where you have uploaded and shown the .head() of each of your data sets.  For each of the proposed datasets, answer the following questions:

## **Dataset #1**


In [1]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [2]:
# Load your data & imports
import numpy as np
import pandas as pd

In [4]:
df = pd.read_csv("/content/drive/MyDrive/Coding Dojo/Data Folder/NYPD_Complaint_Data_Current__Year_To_Date_.csv")

In [5]:
# .head() for your dataframe
df.head()

CMPLNT_NUM  ADDR_PCT_CD BORO_NM CMPLNT_FR_DT CMPLNT_FR_TM CMPLNT_TO_DT  \
0   903695881           69     NaN   12/17/2021     22:13:00          NaN   
1   400462399          113     NaN   12/17/2021     06:21:00          NaN   
2   587910690          113     NaN   12/13/2021     20:05:00          NaN   
3   186105368           52   BRONX   12/07/2021     22:49:00          NaN   
4   185325394          113     NaN   12/06/2021     17:25:00          NaN   

  CMPLNT_TO_TM CRM_ATPT_CPTD_CD HADEVELOPT  HOUSING_PSA  ...  \
0          NaN        COMPLETED        NaN          NaN  ...   
1          NaN        COMPLETED        NaN          NaN  ...   
2          NaN        COMPLETED        NaN          NaN  ...   
3          NaN        COMPLETED        NaN          NaN  ...   
4          NaN        COMPLETED        NaN          NaN  ...   

   TRANSIT_DISTRICT VIC_AGE_GROUP  VIC_RACE VIC_SEX X_COORD_CD Y_COORD_CD  \
0               NaN         25-44     BLACK       M    1011203     174515   
1               NaN         25-44     BLACK       F    1043252     187998   
2               NaN         25-44     BLACK       M    1042087     190443   
3               NaN       UNKNOWN   UNKNOWN       E    1017088     260895   
4               NaN         25-44     BLACK       M    1046176     193100   

    Latitude  Longitude                                   Lat_Lon  \
0  40.645647 -73.902876   (40.64564719600002, -73.90287588699994)   
1  40.682499 -73.787269  (40.682499421000045, -73.78726915499993)   
2  40.689218 -73.791449  (40.689218058000044, -73.79144856999993)   
3  40.882718 -73.881247   (40.88271780700006, -73.88124724999993)   
4  40.696483 -73.776680   (40.69648322200004, -73.77667979199998)   

                        New Georeferenced Column  
0   POINT (-73.90287588699994 40.64564719600002)  
1  POINT (-73.78726915499993 40.682499421000045)  
2  POINT (-73.79144856999993 40.689218058000044)  
3   POINT (-73.88124724999993 40.88271780700006)  
4   POINT (-73.77667979199998 40.69648322200004)  

[5 rows x 36 columns]

In [6]:
df.shape

(449506, 36)

## Questions

### 1) source of your data

The data comes from the New York City Police Department.

https://data.cityofnewyork.us/Public-Safety/NYPD-Complaint-Data-Current-Year-To-Date-/5uac-w243


### 2) Brief description of data

The data includes report information about crimes reported to the NYPD. (Location: Coordinates and Descriptive; Day and Time; Suspect Profile Information; Victim Profile Information; Weather; and several others)

### 3) What is the target?

The target is `"LAW_CAT_CD"`, which is the level of offense (felony, misdemeanor, violation).

###  4) Is this a classification or regression problem?

This will be a classification problem.

### 5) How many features?
Although the dataset has 35 features after being separated from the target, many of them are unnecessary and should be removed before any modeling is done.

For example, there are multiple location features. Two pair of which express the same coordinate information in two different coordiante systems.

There are also several features that are codes which ways of encoding report information and should not be used. They do not predict the target. They were explictly created by the reporting system after the crime was reported and cannot be used to predict future events (they postdict).

A thourough examination of the data will be required to determine all the features that should be removed before modeling.


### 6) How many rows of data.

The dataset has 449,506 rows of data

### 7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

There are several columns which have a lot of missing values. Some missing values may be acceptable (`"PARKS_NM"` has no value if the occurance did not happen in a park), and can be accomadated.

And as mentioned above in question 5, several of the columns should not be used because they may have values that are determined by the target values. Determining which ones may take some research.

Some columns (such as some of the description columns) have multiple values within a single cell. It may be necessary to break them apart into their component values.

## **Dataset #2**

In [ ]:
# Load your data & imports




In [ ]:
# .head() for your dataframe

### Questions

1) # source of your data

2) Brief description of data

3) What is the target?

4) Is this a classification or regression problem?

5) How many features?

6) How many rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?